runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -119,20 +124,42 @@ public Object getPayload(CorfuRuntime runtime) {
     @Override
     public synchronized void releaseBuffer() {
         if (serializedCache != null) {
-            serializedCache.release();
-            if (serializedCache.refCnt() == 0) {
+            serializedCache.buffer.release();
+            if (serializedCache.buffer.refCnt() == 0) {
                 serializedCache = null;
             }
         }
     }
 
     @Override
-    public synchronized void acquireBuffer() {
+    public synchronized void acquireBuffer(boolean metadata) {
         if (serializedCache == null) {
-            serializedCache = Unpooled.buffer();
-            doSerializeInternal(serializedCache);
+            acquireBufferInternal(metadata);
         } else {
-            serializedCache.retain();
+            if (metadata) {
+                serializedCache.buffer.resetReaderIndex();
+                serializedCache.buffer.writerIndex(serializedCache.metadataOffset);
+                doSerializeMetadataInternal(serializedCache.buffer);
```

Comment:
You can slice the buffer at the `metadataOffset` offset

Reply:
Slice will share the same buffer and create a new buffer, I think just set the writerIndex might be simpler?